src/Util/DateTimeFormat.php

```diff
@@ -173,6 +175,29 @@ public static function convert($s = 'now', $tz_to = 'UTC', $tz_from = 'UTC', $fo
 		return $d->format($format);
 	}
 
+	/**
+	 * Fix weird date formats
+	 *
+	 * @param string $dateString
+	 * @return string
+	 */
+	private static function fixDateFormat(string $dateString): string
+	{
+		$pattern =
+			[
+				['#(\w+), (\d+/\d+/\d+) - (\d+:\d+)#', '$1, $2 $3'],
+				['#(\d+-\d+-\d+)T(\d+:\d+:\d+)ZZ#', '$1T$2Z'],
+				['#(\d+-\d+-\d+)T(\d+:\d+:\d+\.\d+)ZZ#', '$1T$2Z'],
+				['#(\w+), (\d+ \w+ \d+) (\d+:\d+:\d+) (.+)#', '$2 $3 $4'],
+				['#(\d+:\d+) (\w+), (\w+) (\d+), (\d+)#', '$1 $2 $3 $4 $5'],
+			];
+
+		foreach ($pattern as $search_replace) {
+			$dateString = preg_replace($search_replace[0], $search_replace[1], $dateString);
+		}
+		return $dateString;
```

Comment:
No need for 2 indentation levels + various minor improvements.
```suggestion
		$patterns = [
			['#(\w+), (\d+/\d+/\d+) - (\d+:\d+)#', '$1, $2 $3'],
			['#(\d+-\d+-\d+)T(\d+:\d+:\d+)ZZ#', '$1T$2Z'],
			['#(\d+-\d+-\d+)T(\d+:\d+:\d+\.\d+)ZZ#', '$1T$2Z'],
			['#(\w+), (\d+ \w+ \d+) (\d+:\d+:\d+) (.+)#', '$2 $3 $4'],
			['#(\d+:\d+) (\w+), (\w+) (\d+), (\d+)#', '$1 $2 $3 $4 $5'],
		];

		foreach ($patterns as $pattern) {
			$dateString = preg_replace($pattern[0], $pattern[1], $dateString);
		}
		
		return $dateString;
```